TypeCobol/Analysis/QualityAnalyzerBase.cs

```diff
@@ -13,23 +13,24 @@ namespace TypeCobol.Analysis
     /// </summary>
     public abstract class QualityAnalyzerBase : IQualityAnalyzer
     {
-        private List<Diagnostic> _diagnostics;
-
         /// <summary>
         /// Unique text identifier of this analyzer.
         /// </summary>
         public string Identifier { get; }
 
-        protected QualityAnalyzerBase([NotNull] string identifier)
-        {
-            Identifier = identifier;
-        }
+        private List<Diagnostic> _diagnostics;
+        protected List<Diagnostic> DiagnosticList => _diagnostics ?? (_diagnostics = new List<Diagnostic>());
 
         /// <summary>
         /// Returns the diagnostics produced by this analyzer.
         /// May be empty, but not null.
         /// </summary>
-        public IEnumerable<Diagnostic> Diagnostics => _diagnostics ?? Enumerable.Empty<Diagnostic>();
+        public IEnumerable<Diagnostic> Diagnostics => DiagnosticList.AsEnumerable();
```

Comment:
`AsEnumerable()` call does nothing here. It won't protect against downcasting as it returns the same instance. You can return `DiagnosticList`, it's ok.